frontend/src/views/AddAppPermission/utils.ts

```diff
@@ -42,6 +51,14 @@ const defaultFormData: AppPermissionFormType = {
     forestClients: [],
     role: null,
     sendUserEmail: false,
+    forestClientInput: {
+        id: "forest-client-number-input",
+        value: "",
+        isValid: true,
+        errorMsg: "",
+        isVerifying: false,
+    },
```

Comment:
The text input for client number is a special case, it doesn't add anything to the form data but it has error and verifying logic. i added this inside formData to prevent prop drilling, for example, role switching is disabled if we are verifying a client number, this requires the role table to have access to the verifying state. I tried to put this logic under the form validation but it's quite limited.